packages/end-to-end/cypress/e2e/tasks.js

```diff
@@ -51,88 +68,59 @@ describe('Tasks', () => {
     // Confirm that location exists
-    cy.get('[data-cy=home-farmButton]')
-      .should('exist')
-      .and('not.be.disabled')
-      .click({ force: true });
-    cy.get('[data-cy=navbar-option]')
-      .eq(1)
+    cy.contains(translation['MENU']['MAP'])
       .should('exist')
       .and('not.be.disabled')
       .click({ force: true });
     cy.contains('First Field').should('be.visible');
 
-    // Add a crop variety
-    cy.get('[data-cy=navbar-hamburger]').should('exist').click();
-    cy.contains(translation['SLIDE_MENU']['TASKS']).should('exist').click();
-    cy.contains(translation['TASK']['ADD_TASK']).should('exist').and('not.be.disabled');
-    cy.visit('/');
-
-    cy.get('[data-cy=home-taskButton]').should('exist').and('not.be.disabled').click();
-
-    cy.contains(translation['TASK']['ADD_TASK']).should('exist').and('not.be.disabled');
-    cy.visit('/');
-  });
-
-  it('CreateCleanTask', () => {
-    // Confirm that location exists
-    cy.get('[data-cy=home-farmButton]')
-      .should('exist')
-      .and('not.be.disabled')
-      .click({ force: true });
-    cy.get('[data-cy=navbar-option]')
-      .eq(1)
-      .should('exist')
-      .and('not.be.disabled')
-      .click({ force: true });
-    cy.contains('First Field').should('be.visible');
-
-    cy.get('[data-cy=navbar-hamburger]').should('exist').click();
-    cy.contains(translation['SLIDE_MENU']['TASKS']).should('exist').click();
+    cy.contains(translation['MENU']['TASKS']).should('exist').click();
     cy.waitForReact();
 
     cy.contains(translation['TASK']['ADD_TASK']).should('exist').and('not.be.disabled').click();
     cy.waitForReact();
-    cy.get('[data-cy=task-selection]').contains(tasks['CLEANING_TASK']).should('exist').click();
+    cy.get(Selectors.TASK_SELECTION).contains(tasks['CLEANING_TASK']).should('exist').click();
 
     //Create an unassigned cleaning task due tomorrow
     const date = new Date();
     date.setDate(date.getDate() + 1);
     const getDateInputFormat = (date) => moment(date).format('YYYY-MM-DD');
     const dueDate = getDateInputFormat(date);
-    cy.get('[data-cy=addTask-taskDate]').should('exist').type(dueDate);
+    cy.get(Selectors.ADD_TASK_TASK_DATE).should('exist').type(dueDate);
 
-    cy.get('[data-cy=addTask-continue]').should('exist').and('not.be.disabled').click();
+    cy.get(Selectors.ADD_TASK_CONTINUE).should('exist').and('not.be.disabled').click();
 
     cy.contains('First Field').should('be.visible');
     // eslint-disable-next-line cypress/no-unnecessary-waiting
     cy.wait(500, { log: false });
-    cy.get('[data-cy=map-selectLocation]').click({ force: false });
+    cy.get(Selectors.MAP_SELECT_LOCATION).click({ force: false });
+
+    cy.get(Selectors.ADD_TASK_LOCATION_CONTINUE).should('exist').and('not.be.disabled').click();
+
+    // This screen only present when this test file is run after crops.js
+    cy.get(Selectors.ADD_TASK_CROPS_CONTINUE).should('exist').and('not.be.disabled').click();
```

Comment:
This is the cause of the temporal dependency between `crops.js` and `tasks.js`. There is an additional page in this multi-page form when a crop plan exists on the farm.

Reply:
In the future to make this more robust and ensure the tests are independent, we'll likely need to bake in the creation of a crop in the tasks test itself as a part of the test setup before assertions. We'll probably end up abstracting the code that does the crop creation on its own reusable command

Reply:
Ah that's a good idea! I was thinking the spec needed to be more flexible here, but instead the environment could have been made more predictable.